### PR TITLE
Fix active responses timeout to take into account alert details

### PR DESCRIPTION
**Pull request by TomasTurina**
|Related issue|
|---|
|https://github.com/wazuh/wazuh/issues/9370|

## Description

Let the AR script extract the keys it needs from the alert and let `wazuh-execd` know which are their values, so then `wazuh-execd` can determine if this AR was already executed or not for this alert:

- If the AR for this alert is in the timeout list, `wazuh-execd` tells the AR script to abort its execution.
- If the AR for this alert isn't in the timeout list, `wazuh-execd` tells the AR script to continue its execution.

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
  - [x] Windows
  - [x] MAC OS X
- [x] Source installation
- [x] Package installation
- [x] Source upgrade
- [x] Package upgrade
- [x] Review logs syntax and correct language

- Memory tests for Linux
  - [x] Scan-build report
  - [x] Coverity
  - [x] Valgrind (memcheck and descriptor leaks check)

- [x] Added unit tests (for new features)
- [ ] Stress test for affected components

### CODE REVIEW

File: src/os_execd/execd.c [hunk 1]
```diff
@@ -450,94 +458,61 @@ STATIC void ExecdStart(int q)
...
                 continue;
             }
 
+            /* Set rkey initially with the name of the AR */
+            memset(rkey, '\0', OS_SIZE_4096);
+            snprintf(rkey, OS_SIZE_4096 - 1, "%s", basename_ex(cmd[0]));
+
+            keys_json = get_json_from_input(buffer);
+            if (keys_json != NULL) {
+	            char *action = get_command_from_json(keys_json);
```
Comment (Dooque) [inline src/os_execd/execd.c:498]: const

File: src/os_execd/execd.c [hunk 2]
```diff
@@ -450,94 +458,61 @@ STATIC void ExecdStart(int q)
...
+                if ((action != NULL) && (strcmp(CHECK_KEYS_ENTRY, action) == 0)) {
+                    char *keys = get_keys_from_json(keys_json);
```
Comment (Dooque) [inline src/os_execd/execd.c:500]: const

File: src/os_execd/win_execd.c [hunk 3]
```diff
@@ -195,78 +200,128 @@ void WinExecdRun(char *exec_msg)
...
+    cmd_parameters = cJSON_PrintUnformatted(json_root);
+
+    /* Execute command */
+    mdebug1("Executing command '%s %s'", cmd[0], cmd_parameters ? cmd_parameters : "");
+
+    wfd_t *wfd = wpopenv(cmd[0], cmd, W_BIND_STDIN | W_BIND_STDOUT);
+    if (wfd) {
+        char buffer[OS_SIZE_4096];
+        char rkey[OS_SIZE_4096];
+        cJSON *keys_json = NULL;
```
Comment (Dooque) [inline src/os_execd/win_execd.c:218]: const

File: src/os_execd/win_execd.c [hunk 4]
```diff
@@ -195,78 +200,128 @@ void WinExecdRun(char *exec_msg)
...
             return;
         }
 
+        /* Set rkey initially with the name of the AR */
+        memset(rkey, '\0', OS_SIZE_4096);
+        snprintf(rkey, OS_SIZE_4096 - 1, "%s", basename_ex(cmd[0]));
+
+        keys_json = get_json_from_input(buffer);
+        if (keys_json != NULL) {
+            char *action = get_command_from_json(keys_json);
```
Comment (Dooque) [inline src/os_execd/win_execd.c:240]: const

File: src/os_execd/win_execd.c [hunk 5]
```diff
@@ -195,78 +200,128 @@ void WinExecdRun(char *exec_msg)
...
+            if ((action != NULL) && (strcmp(CHECK_KEYS_ENTRY, action) == 0)) {
+                char *keys = get_keys_from_json(keys_json);
```
Comment (Dooque) [inline src/os_execd/win_execd.c:242]: const

**Review by Dooque — COMMENTED** [review]
In general try to fix const correctness.

File: src/active-response/active_responses.c [hunk 6]
```diff
@@ -9,7 +9,15 @@
...
+ * Build JSON message with keys to be sent to execd
+ * @param ar_name Name of active response
+ * @param keys Array of keys
+ * @return char * with the JSON message in string format
+ */
+static char* build_json_keys_message(const char *ar_name, char **keys);
+
+void write_debug_file(const char *ar_name, const char *msg) {
     char path[PATH_MAX];
     char *timestamp = w_get_timestamp(time(NULL));
```
[the remark this answers is not in the document]
Comment (TomasTurina) [reply src/active-response/active_responses.c:23]: You are right! Will change this.

File: src/active-response/active_responses.h [hunk 7]
```diff
@@ -14,61 +14,100 @@
...
-void write_debug_file (const char *ar_name, const char *msg);
+void write_debug_file(const char *ar_name, const char *msg);
+
+/**
+ * @brief Set wazuh home directory and check message from stdin
+ * @param argv Arguments of the script
+ * @param message JSON message from stdin
+ * @return Command from message
+ */
+int setup_and_check_message(char **argv, cJSON **message);
```
[the remark this answers is not in the document]
Comment (TomasTurina) [reply src/active-response/active_responses.h:53]: This won't have effect. This warning is generated:
```
active-response/active_responses.h:53:1: warning: type qualifiers ignored on function return type [-Wignored-qualifiers]
   53 | const int setup_and_check_message(char **argv, cJSON **message)
```

File: src/active-response/active_responses.h [hunk 8]
```diff
@@ -14,61 +14,100 @@
...
+
+/**
+ * @brief Send message with keys and check message from stdin
+ * @param argv Arguments of the script
+ * @param keys Keys to be sent
+ * @return Command from message
+ */
+int send_keys_and_check_message(char **argv, char **keys);
```
[the remark this answers is not in the document]
Comment (TomasTurina) [reply src/active-response/active_responses.h:61]: This won't have effect. This warning is generated:
```
active-response/active_responses.h:61:1: warning: type qualifiers ignored on function return type [-Wignored-qualifiers]
   61 | const int send_keys_and_check_message(char **argv, char **keys);
```

File: src/os_execd/win_execd.c (hunk 3 above)
Comment (TomasTurina) [reply src/os_execd/win_execd.c:222]: This can't be const since it is modified below.

**Review by vikman90 — APPROVED** [review]

**Pull request merged**